engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1027,6 +1027,19 @@ public int processEvents()
         }
       }
       o.stats.lastWindowedStats = stats;
+      if (!o.stats.responses.isEmpty()) {
+        if (o.stats.operatorResponses == null) {
+          o.stats.operatorResponses = new ArrayList<>();
+        } else {
+          o.stats.operatorResponses.clear();
+        }
+        StatsListener.OperatorResponse operatorResponse;
+        while ((operatorResponse = o.stats.responses.poll()) != null) {
+          o.stats.operatorResponses.add(operatorResponse);
+        }
+      } else {
+        o.stats.operatorResponses = null;
```

Comment:
That's what I thought the intention was, but in this case benefits are negligible. Why do we set it to null in the else block? To avoid more if then else clutter? ;-)